Using TARGET_OS_* macros without checking if they are defined is an
error in clang 11.

--- nss/gtests/google_test/gtest/include/gtest/internal/gtest-port-arch.h.orig	2023-12-19 04:23:08
+++ nss/gtests/google_test/gtest/include/gtest/internal/gtest-port-arch.h	2024-01-17 13:13:52
@@ -69,7 +69,7 @@
 #elif defined __APPLE__
 # define GTEST_OS_MAC 1
 # include <TargetConditionals.h>
-# if TARGET_OS_IPHONE
+# if defined(TARGET_OS_IPHONE) && TARGET_OS_IPHONE
 #  define GTEST_OS_IOS 1
 # endif
 #elif defined __DragonFly__
--- nss/lib/freebl/unix_rand.c.orig	2023-12-19 04:23:08
+++ nss/lib/freebl/unix_rand.c	2024-01-17 13:17:16
@@ -594,7 +594,7 @@ RNG_GetNoise(void *buf, size_t maxbytes)
 
 #ifdef DARWIN
 #include <TargetConditionals.h>
-#if !TARGET_OS_IPHONE
+#if !defined(TARGET_OS_IPHONE) || !TARGET_OS_IPHONE
 #include <crt_externs.h>
 #endif
 #endif
@@ -607,7 +607,7 @@ RNG_SystemInfoForRNG(void)
     const char *const *cp;
     char *randfile;
 #ifdef DARWIN
-#if TARGET_OS_IPHONE
+#if defined(TARGET_OS_IPHONE) && TARGET_OS_IPHONE
     /* iOS does not expose a way to access environ. */
     char **environ = NULL;
 #else
